### ISSUE TITLE
srcmap-runtime is inaccurate

### ISSUE THREAD

**Issue by duytai** [ethereum/remix-ide#2847]
## Description

srcmap-runtime is inaccurate

## Environment

- Compiler version:  0.5.10+commit.5a6ea5b1.Linux.g++
- Target EVM version (as per compiler settings):
- Framework/IDE (e.g. Truffle or Remix):
- EVM execution environment / backend / blockchain client:
- Operating system: Ubuntu

## Steps to Reproduce
I use the below command to get get srcmap-runtime of smart contract __RampInstantTokenPool__(https://etherscan.io/address/0x85e112e0f75f2b4751a647b81fcbca3bb3c78f84#code)
```bash
solc --combined-json bin-runtime,srcmap-runtime RampInstantTokenPool.sol
```
After that, I decompress srcmap-runtime and try to print out all mappings from bytecode to source code. Some mapped code are incomplete. For example:
```
equire(msg.sender == owner, "only the owner can call this");
```
It must be 
```
require(msg.sender == owner, "only the owner can call this");
```

here is decompress srcmap code snippet:
```javascript
decompress(srcmap) {
    return srcmap
      .split(';')
      .reduce((result, slfj) => {
        const [s, l, f, j] = slfj.split(':')
        const [last] = result.slice(-1)
        return result.concat({
          s: s ? parseInt(s, 10) : last.s,
          l: l ? parseInt(l, 10) : last.l,
          f: f ? parseInt(f, 10) : last.f,
          j: j || last.j,
        })
      }, [])
  }
```

**Comment by chriseth** [comment]
It seems that remix uses character counts for source references instead of byte amounts - is that correct?

**Comment by yann300** [comment]
yes we use character counts, we could work with bytes but not sure if the editor is directly handling it and if not we would need to process and parse the source I guess.

**Comment by chriseth** [comment]
I still think byte count is the better metric because it is independent of the encoding (which is not really specified) and all the weird unicode effects.
